java/api/src/main/java/org/ray/api/id/ObjectId.java

```diff
@@ -39,12 +50,66 @@ private static ObjectId genNil() {
   /**
    * Generate an ObjectId with random value.
    */
-  public static ObjectId randomId() {
+  public static ObjectId fromRandom() {
     byte[] b = new byte[LENGTH];
     new Random().nextBytes(b);
     return new ObjectId(b);
   }
 
+  /**
+   * Compute the object ID of an object put by the task.
+   */
+  public static ObjectId forPut(TaskId taskId, int putIndex) {
+    return forPut(taskId, putIndex, TransportType.STANDARD);
+  }
+
+  /**
+   * Compute the object ID of an object put by the task.
+   */
+  public static ObjectId forPut(TaskId taskId, int putIndex, TransportType transportType) {
```

Comment:
I don't think `transportType` needs to be exposed in Java. 
And after Java worker is migrated to the core worker, other flags don't need as well.